File: 0217-contains-duplicate/0217-contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        int n = nums.size();
        unordered_map<int, bool> mp; //tc o(1) for the hash set operation
        for(int i=0; i<n; i++){
            if(mp.count(nums[i]) > 0){
                return true;
            }
            mp[nums[i]] = true;
        }
        return false;
    }
};

File: 0217-contains-duplicate/0217-contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        #this code has a time complexity of O(n)
        #return len(nums) != len(set(nums))
        #to optimse the space complexity
        
        seen = set()
        for n in nums:
            if n in seen:
                return True
            seen.add(n)
        return False